### PR TITLE
Updated Dependency Injection

### PR DESCRIPTION
**Pull request by ckaplan4**
- Removed Accessors.Shared, Common.Shared, Managers.Shared, Engines.Shared projects
- Removed Factory files 
- Added TestManager, TestEngine, TestAccessor and setup constructor injection for these services 
- Added unit tests for TestManager, TestEngine, TestAccessor

### CODE REVIEW

File: templates/dplsln/DPL.Template.Common.Contracts/Logger.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace DPL.Template.Common
+{
+    public class Logger
```
Comment (bbarber) [inline templates/dplsln/DPL.Template.Common.Contracts/Logger.cs:5]: I see this duplicated in two different places. Is this expected? If not, which should survive?
image

File: templates/dplsln/DPL.Template.Common.Contracts/ServiceContractBase.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using DPL.Template.Common.Contracts;
+
+namespace DPL.Template.Common
+{
+    public abstract class ServiceContractBase
```
Comment (bbarber) [inline templates/dplsln/DPL.Template.Common.Contracts/ServiceContractBase.cs:6]: I see this duplicated. Was this expected? If not, which should be removed?
image

File: templates/dplsln/DPL.Template.Common.Contracts/Logger.cs (hunk 1 above)
Comment (bbarber) [reply templates/dplsln/DPL.Template.Common.Contracts/Logger.cs:5]: "I think common shared project needs to be deleted. Chad wanted me to get rid of all the .shared projects. Missed that one" -Ceren

(Same for the other duplicated files)

I will handle removing these files in this PR

Comment (bbarber) [reply templates/dplsln/DPL.Template.Common.Contracts/Logger.cs:5]: Files removed. Resolving

**Review by twilsman — APPROVED** [review]

**Pull request merged**